### PR TITLE
dnn: updated yolov4-tiny hash

### PR DESCRIPTION
**Pull request by mshabunin**
Merge with opencv/opencv#23008

Looks like this model has been changed in the source - updated file hash

### CODE REVIEW

File: testdata/dnn/download_models.py [hunk 1]
```diff
@@ -905,7 +905,7 @@ def get_confirm_token(response):  # in case of large files
     Model(
         name='YOLOv4-tiny',  # https://github.com/opencv/opencv/issues/17148
         url='https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4-tiny.weights',
-        sha='d110379b7b86899226b591ad4affc7115f707157',
+        sha='451caaab22fb9831aa1a5ee9b5ba74a35ffa5dcb',
```
Comment (alalek) [inline testdata/dnn/download_models.py:908]: We don't override files created by this script (there is 3.4 branch which expects old sha / and tested with old file only).
It should be a new file (with date suffix `-YYYY-MM`). Also we need to update tests.

There are 2 links on the release page for `yolov4-tiny.weights`:
- https://github.com/AlexeyAB/darknet/releases/download/yolov4/yolov4-tiny.weights
- https://github.com/AlexeyAB/darknet/releases/download/darknet_yolo_v4_pre/yolov4-tiny.weights

But both with the same content (`sha1:451caaab22fb9831aa1a5ee9b5ba74a35ffa5dcb`).

BTW, I have a copy of old file with download date 2020-07-03 (`sha1:d110379b7b86899226b591ad4affc7115f707157`).

Comment (mshabunin) [reply testdata/dnn/download_models.py:908]: I've replaced old record with a new one and updated all tests.

Comment (alalek) [reply testdata/dnn/download_models.py:908]: According to dates on this page suffix should be `-2020-12` (15 Dec 2020): https://github.com/AlexeyAB/darknet/releases

Comment (alalek) [reply testdata/dnn/download_models.py:908]: `darknet_yolo_v4_pre` -> `yolov4` in URL?

Comment (mshabunin) [reply testdata/dnn/download_models.py:908]: Updated.
Also changed `yolov4` URL from Google Drive to GitHub release, the file has same SHA1.

Comment (alalek) [reply testdata/dnn/download_models.py:908]: `yolov4-tiny-2022-12.cfg` -> `2020-12` too

**Pull request merged**